docs/TEMPORARY_DIRECTORIES.md

```diff
@@ -0,0 +1,201 @@
+those in `/var/tmp` after 30 days.
+
+This automatic clean-up is important to ensure disk usage of these temporary
+directories doesn't grow without bounds, even when programs abort abnormally or
+otherwise don't clean up the temporary files/directories they create. On the
+other hand it creates problems for long-running software that does not expect
+temporary files it operates on are suddenly removed. There are a couple of
+strategies to avoid these issues:
+
+1. Make sure to always keep a file descriptor to the temporary files you
+   operate on open, and only access the files through them. This way it doesn't
+   matter whether the filenames have been unlinked from the file system: as
+   long as you have the file descriptor open you can still access the file for
+   both reading and writing. When operating this way it is recommended to delete
+   the files right after creating them to ensure that on abnormal program
+   termination the files or directories are implicitly released by the kernel.
+
+2. 🥇 Use `memfd_create()` or `O_TMPFILE`. This is an extension of the suggestion
+   above: files created this way are never linked under a filename in the file
+   system. This means they are not subject to ageing (as they come unlinked out
+   of the box), and there's no time window where the file exists but not yet
+   deleted, and thus behaviour is fully robust towards abnormal program
+   termination as there are never files on disk that need to be explicitly
+   deleted.
+
+3. 🥇 Operate below a sub-directory of `/tmp/` and `/var/tmp/` you created, and
+   take a BSD file lock ([`flock(dir_fd,
+   LOCK_SH)`](http://man7.org/linux/man-pages/man2/flock.2.html)) on that
+   sub-directory. This is particularly interesting when operating on more than a
+   single file, or on file nodes that are not plain regular files, for example
+   when extracting a tarball to a temporary directory. The ageing algorithm
+   will skip all directories (and everything below them) that are locked
+   through a BSD file lock. As BSD file locks are automatically released when
+   the file descriptor they are taken on is closed, and all file descriptors
+   opened by a process are implicitly closed when it exits this is a robust
+   mechanism that ensures all generated temporary files are subject to ageing
+   when the program that owns them abnormally dies, but not while it is still
+   running. This is a particularly interesting option when decompressing
+   tarballs that contain files with old modification/access times, as extracted
+   files are otherwise immediately candidates for deletion by the ageing
+   algorithm. The [`flock`](http://man7.org/linux/man-pages/man1/flock.1.html)
+   tool of the `util-linux` packages makes this concept available to shell
+   scripts. Note that `systemd-tmpfiles` only checks for BSD file locks on
+   directories, locks on other types of file nodes (including regular files)
+   are not considered.
```

Comment:
So, I thought about this, but I decided against checking for these locks on files for implementation reasons: it would mean we'd have to open() all files in order to check the lock, and that would make things a lot slower and would trigger atime updates we probably want to avoid, since we care about atime as input. OTOH the dirs we open anyway, hence the lock checking is basically free there anyway.

We might reconsider that one day, but it appeared unnecessary to me at least for the beginning, and with a risk of generating problems and making things avoidably slower...

But I guess that discussion should be had in #11482.